File: requery-test/src/main/java/io/requery/test/immutables/Person.java
package io.requery.test.immutables;

import io.requery.Column;
import io.requery.Entity;
import io.requery.Generated;
import io.requery.Key;
import org.immutables.value.Value;

import java.util.Date;
import java.util.UUID;

@Value.Immutable
@Entity(model = "immutable", builder = ImmutablePerson.Builder.class)
public abstract class Person {

    @Key @Generated
    public abstract int getId();

    public abstract String getName();
    public abstract String getEmail();
    public abstract Date getBirthday();
    public abstract int getAge();

    public abstract String getAbout();

    @Value.Default
    @Column(unique = true)
    public UUID getUUID() {
        return UUID.randomUUID();
    }
}

File: requery-test/kotlin-test/src/main/kotlin/io/requery/test/kt/data/Person.kt
package io.requery.test.kt.data

import io.requery.*
import java.net.URL
import java.util.Date
import java.util.UUID

@Entity(model = "ktdata")
data class Person (

    @get:Key
    var id: Int,
    var name: String,
    var email: String,
    var birthday: Date,
    var age: Int,

    var about: String,

    @get:Column(unique = true)
    val uuid: UUID,
    val homepage: URL,
    val picture: String
)
